mitodl/open-discussions
# Infinite Corridor front page

#### Pre-Flight checklist


#### What are the relevant tickets?
#3613 

#### What's this PR do?
This PR sets up a front page, router, and some jest-related testing-utils.

#### How should this be manually tested?
1. Create a few fields if you don't have any. Do this via the form at http://yourname.od.odl.local:8063/api/v0/fields as a superuser. 
2. Visit http://yourname.od.odl.local:8063/infinite and you should see a list of the fields you created.
3. Each field should link to a placeholder page.
4. Back at the front page, submitting text in the searchbox should send you to the search page with the correct query param.

#### Any background context you want to provide?
⚠️ This PR does add a new dependency, [react-query](https://www.npmjs.com/package/react-query). 

I have heard us (Alice before she left; @rhysyngsun via https://github.com/mitodl/ol-npm-libraries#best-practices, possibly via discussions, too) say that this is the direction we should move, though I'm not sure to what extent we've actually used this library before in our apps.

See https://github.com/mitodl/hq/discussions/153 for more, but its usage is basically:
```ts
// status is one of "loading", "error", or "success"
const { data, status, ...lotsOfOtherProperties } = useQuery(queryKey, queryFn, options)
```
where `queryFn` is a function that returns a promise resolving to `data`. We could use any data-fetching tool (native fetch; axios, superagent, etc) to write the `queryFn` component. 

**Update:** Now using axios.
